lighthouse-treemap/app/src/main.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env browser */
+
+/* globals webtreemap Util */
+
+/**
+ * Allows for saving the document and loading with data intact.
+ */
+function injectOptions() {
+  // @ts-expect-error
+  if (!window.__injected) return;
+
+  const scriptEl = document.createElement('script');
+  scriptEl.innerHTML = `
+    window.__TREEMAP_OPTIONS = ${JSON.stringify(window.__TREEMAP_OPTIONS)};
+    window.__injected = true;
+  `;
+  document.head.append(scriptEl);
+}
+
+/**
+ * @param {Treemap.Options} options
+ */
+function init(options) {
+  // ==== temporary
+  Util.find('main').textContent = JSON.stringify(options.lhr);
+  // eslint-disable-next-line no-console
+  console.log({webtreemap});
+  // ==== temporary
+
+  window.__TREEMAP_OPTIONS = options;
+  injectOptions();
+
+  if (window.ga) {
+    // TODO what are these?
+    // window.ga('send', 'event', 'treemap', 'open in viewer');
+    window.ga('send', 'event', 'report', 'open in viewer');
+  }
+
+  // eslint-disable-next-line no-console
+  console.log('window.__TREEMAP_OPTIONS', window.__TREEMAP_OPTIONS);
+}
+
+/**
+ * @param {string} message
+ */
+function showError(message) {
+  document.body.textContent = message;
+}
+
+async function main() {
+  if (window.__TREEMAP_OPTIONS) {
+    init(window.__TREEMAP_OPTIONS);
```

Comment:
```suggestion
    // Prefer the hardcoded options from a saved HTML file above all.
    init(window.__TREEMAP_OPTIONS);
```

is this the only situation that this applies?